### PR TITLE
`Development`: Add test exam test run e2e test

### PR DESCRIPTION
**Pull request by TheZoker**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

Test exams support test runs, where a instructor can test the exam before conducting it. 

### Description

This PR adds e2e tests for the test run within test exams. We create a test run, change the working time, conduct it and delete it. 

#### Code Review
- [ ] Review 1
- [ ] Review 2

### CODE REVIEW

**Review by mertyldrr — DISMISSED** [review]
Looks good to me 👍

File: src/test/cypress/support/constants.ts [hunk 1]
```diff
@@ -17,8 +17,8 @@ export const TIME_FORMAT = 'YYYY-MM-DDTHH:mm:ss.SSS';
 
 // Exercise types
 export enum EXERCISE_TYPE {
-    Text,
-    Modeling,
-    Quiz,
-    Programming,
+    Text = 'text',
```
Comment (rriyaldhi) [inline src/test/cypress/support/constants.ts:20]: Just curious, why is this change necessary?

**Review by rriyaldhi — COMMENTED** [review]
Change looks good. Just have 1 small question.

File: src/test/cypress/support/constants.ts (hunk 1 above)
Comment (TheZoker) [reply src/test/cypress/support/constants.ts]: These were commits included in the old PR, I now rebased it :)

File: src/test/cypress/e2e/exam/test-exam/TestExamTestRun.cy.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        examTestRun.createTestRun();
+        examTestRun.setWorkingTimeMinutes(minutes);
+        examTestRun.setWorkingTimeSeconds(seconds);
+        examTestRun.confirmTestRun().then((testRunResponse: Interception) => {
+            const testRun = testRunResponse.response!.body;
+            testRunID = testRun.id;
+
+            expect(testRunResponse.response!.statusCode).to.eq(200);
+            expect(testRun.testRun).to.eq(true);
+            expect(testRun.submitted).to.eq(false);
```
Comment (bassner) [inline src/test/cypress/e2e/exam/test-exam/TestExamTestRun.cy.ts:82]: (optional) Is this maybe better?

```suggestion
            expect(testRun.testRun).to.be.true;
            expect(testRun.submitted).to.be.false;
```

File: src/test/cypress/e2e/exam/test-exam/TestExamTestRun.cy.ts [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+                examExerciseGroupCreation.addGroupWithExercise(exerciseArray, exam, EXERCISE_TYPE.Quiz, { quizExerciseID: 0 });
+
+                examExerciseGroupCreation.addGroupWithExercise(exerciseArray, exam, EXERCISE_TYPE.Modeling);
+            });
+        });
+    });
+
+    beforeEach('Create test run instance', () => {
+        cy.login(instructor);
+        // API call does not work yet, for now we use the UI to create the test run
```
Comment (bassner) [inline src/test/cypress/e2e/exam/test-exam/TestExamTestRun.cy.ts:52]: (optional) Is this actually a TODO for the future? :D
```suggestion
        // TODO: API call does not work yet, for now we use the UI to create the test run
```

File: src/test/cypress/e2e/exam/test-exam/TestExamTestRun.cy.ts [hunk 4]
```diff
@@ -0,0 +1,154 @@
+import { Exam } from 'app/entities/exam.model';
+import { CypressExamBuilder, convertCourseAfterMultiPart } from '../../../support/requests/CourseManagementRequests';
+import dayjs from 'dayjs/esm';
+import submission from '../../../fixtures/exercise/programming/build_error/submission.json';
+import { Course } from 'app/entities/course.model';
+import { generateUUID } from '../../../support/utils';
+import { EXERCISE_TYPE } from '../../../support/constants';
+import { Exercise } from 'src/test/cypress/support/pageobjects/exam/ExamParticipation';
```
Comment (bassner) [inline src/test/cypress/e2e/exam/test-exam/TestExamTestRun.cy.ts:8]: Let's be consistent here! If you access something from the support directory for example, either do a relative or an absolute import, but don't mix it. :)
Ideally, you would decide for one or the other and apply it to all imports.

**Review by bassner — CHANGES_REQUESTED** [review]
Hey! I've added a few small suggestions below

File: src/test/cypress/e2e/exam/test-exam/TestExamTestRun.cy.ts (hunk 2 above)
Comment (TheZoker) [reply src/test/cypress/e2e/exam/test-exam/TestExamTestRun.cy.ts]: Good idea!

File: src/test/cypress/e2e/exam/test-exam/TestExamTestRun.cy.ts (hunk 4 above)
Comment (TheZoker) [reply src/test/cypress/e2e/exam/test-exam/TestExamTestRun.cy.ts]: Good point, the auto import seems to be inconsistent here for some reason. I fixed it for this PR and fixed the other files in another PR, that I will open soon. Thanks for the hint!

Comment (TheZoker) [reply src/test/cypress/e2e/exam/test-exam/TestExamTestRun.cy.ts]: PR is open here: https://github.com/ls1intum/Artemis/pull/6297

**Review by JohannesStoehr — APPROVED** [review]
New test logic looks good to me

**Review by mertyldrr — APPROVED** [review]
Reapprove

**Review by rriyaldhi — APPROVED** [review]
